sequencer/node/node.go

```diff
@@ -319,275 +316,330 @@ func NewNode(mode Mode, cfg *config.Node, version string) (*Node, error) {
-						"verifier.MaxTx = %v, verifier.NLevels = %v",
-						cfg.Coordinator.Circuit.MaxTx, cfg.Coordinator.Circuit.NLevels,
-						verifier.MaxTx, verifier.NLevels,
-					))
-			}
-		}
+	//TODO: Initialize server proofs
+	// serverProofs := make([]prover.Client, len(cfg.Coordinator.ServerProofs.URLs))
+	// for i, serverProofCfg := range cfg.Coordinator.ServerProofs.URLs {
+	// 	serverProofs[i] = prover.NewProofServerClient(serverProofCfg,
+	// 		cfg.Coordinator.ProofServerPollInterval.Duration)
+	// }
 
-		coord, err = coordinator.NewCoordinator(
-			coordinator.Config{
-				ForgerAddress:           cfg.Coordinator.ForgerAddress,
-				ConfirmBlocks:           cfg.Coordinator.ConfirmBlocks,
-				L1BatchTimeoutPerc:      cfg.Coordinator.L1BatchTimeoutPerc,
-				ForgeRetryInterval:      cfg.Coordinator.ForgeRetryInterval.Duration,
-				ForgeDelay:              cfg.Coordinator.ForgeDelay.Duration,
-				MustForgeAtSlotDeadline: cfg.Coordinator.MustForgeAtSlotDeadline,
-				IgnoreSlotCommitment:    cfg.Coordinator.IgnoreSlotCommitment,
-				ForgeOncePerSlotIfTxs:   cfg.Coordinator.ForgeOncePerSlotIfTxs,
-				ForgeNoTxsDelay:         cfg.Coordinator.ForgeNoTxsDelay.Duration,
-				SyncRetryInterval:       cfg.Coordinator.SyncRetryInterval.Duration,
-				PurgeByExtDelInterval:   cfg.Coordinator.PurgeByExtDelInterval.Duration,
-				EthClientAttempts:       cfg.Coordinator.EthClient.Attempts,
-				EthClientAttemptsDelay:  cfg.Coordinator.EthClient.AttemptsDelay.Duration,
-				EthNoReuseNonce:         cfg.Coordinator.EthClient.NoReuseNonce,
-				EthTxResendTimeout:      cfg.Coordinator.EthClient.TxResendTimeout.Duration,
-				MaxGasPrice:             cfg.Coordinator.EthClient.MaxGasPrice,
-				MinGasPrice:             cfg.Coordinator.EthClient.MinGasPrice,
-				GasPriceIncPerc:         cfg.Coordinator.EthClient.GasPriceIncPerc,
-				TxManagerCheckInterval:  cfg.Coordinator.EthClient.CheckLoopInterval.Duration,
-				DebugBatchPath:          cfg.Coordinator.Debug.BatchPath,
-				Purger: coordinator.PurgerCfg{
-					PurgeBatchDelay:      cfg.Coordinator.L2DB.PurgeBatchDelay,
-					InvalidateBatchDelay: cfg.Coordinator.L2DB.InvalidateBatchDelay,
-					PurgeBlockDelay:      cfg.Coordinator.L2DB.PurgeBlockDelay,
-					InvalidateBlockDelay: cfg.Coordinator.L2DB.InvalidateBlockDelay,
-				},
-				ForgeBatchGasCost: cfg.Coordinator.EthClient.ForgeBatchGasCost,
-				VerifierIdx:       uint8(verifierIdx),
-				TxProcessorConfig: txProcessorCfg,
-				ProverReadTimeout: cfg.Coordinator.ProverWaitReadTimeout.Duration,
-			},
-			historyDB,
-			l2DB,
-			txSelector,
-			batchBuilder,
-			nil, //serverProofs
-			client,
-			&scConsts,
-			initSCVars,
-			etherScanService,
+	txProcessorCfg := txprocessor.Config{
+		NLevels: uint32(cfg.Coordinator.Circuit.NLevels),
+		MaxTx:   uint32(cfg.Coordinator.Circuit.MaxTx),
+		ChainID: chainIDU16,
+		// MaxFeeTx: common.RollupConstMaxFeeIdxCoordinator,
+		MaxL1Tx: common.RollupConstMaxL1Tx,
+	}
+	var verifierIdx int
```

Comment:
VerifierIdx is think is to match different proovers (circuit) with the sc, Since we'll have only one instance of proover. I don't think we'll need this.